src/Indicator.vala

```diff
@@ -367,6 +367,8 @@ public class Session.Indicator : Wingpanel.Indicator {
         });
 
         current_dialog.show_all ();
+
+        current_dialog.present ();
```

Comment:
I notice the docs for `Window#present` say to prefer using `present_with_time()` to support focus stealing prevention, but I couldn't figure out how to get a `uint32` timestamp (I'm a web dev by day - I don't have much C-style experience). Would `present` be fine anyway for this case because you don't want anything to take focus priority over this particular dialog?

Reply:
I usually like to search for and steal code from other parts of the elementary repos, for example, searching for `present_with_time` you get multiple usage thus: `active_window.present_with_time (Gdk.CURRENT_TIME);` so you could try using `Gdk.CURRENT_TIME` for that timestamp?

Reply:
I like your style :grin:  I did find `Gdk.CURRENT_TIME` but as far as I can tell that's just a constant value of `0`, which is intended to represent the time _now_, at the point of usage. The `present_with_time` docs advise:

> The timestamp should be gathered when the window was requested to be shown ( when clicking a link for example), rather than once the window is ready to be shown

So I was going to generate the timestamp [at the start of the `show_dialog` function](https://github.com/elementary/wingpanel-indicator-session/blob/bbeec31f7df3f76fd744156237db32ea747764e7/src/Indicator.vala#L317) (or closer to the user interaction if I could) and then use that timestamp in `present_with_time` at the end of the same function.

I imagine practically speaking the difference is negligible, but, in general, passing `0` to `present_with_time` sounds like it defeats the purpose of the function.

But I am new to GTK, and haven't done much desktop programming for about 15 years, so it's very possible I've misunderstood!

Reply:
`present` should actually replace `show_all` here, but yeah just `present` is fine, there's other cases where we get an event timestamp but in this case I don't think we need to really worry about `present_with_time` since you'd get maybe a few miliseconds difference of timestamp and like you said it's okay if this particular dialog is "stealing focus" since it's always on top etc

Reply:
> present should actually replace show_all here

I tried removing `show_all` but this results in the dialog not being shown at all - am I missing something?

Reply:
> I tried removing `show_all` but this results in the dialog not being shown at all - am I missing something?
> …
> Sounds like this call is still needed, to show the contents of the window?

`show.connect (() => get_child ().show_all ());` at the end of the EndSessionDialog construct block should do the trick.